src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/ProjectSystem/RazorProjectInfo.cs

```diff
@@ -12,20 +12,20 @@
 
 namespace Microsoft.AspNetCore.Razor.ProjectSystem;
 
-internal sealed class RazorProjectInfo
+internal sealed record class RazorProjectInfo
```

Comment:
Does making this a record help in some way? Note that the equality of this record won't be correct because it will need to use `SequenceEquals` for the `Documents` property.

Reply:
It made it cleaner to add a `with` for modifying readonly properties than having to write out `new(...)`